lib/active_support/deprecation_test_helper.rb

```diff
@@ -1,10 +1,82 @@
 # frozen_string_literal: true
 
 require "active_support/deprecation_test_helper/version"
+require 'active_support/deprecation'
 
 module ActiveSupport
   module DeprecationTestHelper
-    class Error < StandardError; end
-    # Your code goes here...
+    class << self
+      def configure
+        defined?(Minitest) || defined?(RSpec) or raise 'Unexpected test suite encountered. This only supports Rspec and Minitest'
+
+        ActiveSupport::Deprecation.include(self)
+
+        if defined?(Minitest)
+          Minitest.after_run(after_all_callback)
+        elsif defined?(RSpec)
+          RSpec.configuration.after(:all, after_all_callback)
+        end
+      end
+
+      def reset
+        @allowed_warnings    = Set.new
+        @unexpected_warnings = Set.new
+      end
+
+      def allow_warning(message_or_regex)
+        unless message_or_regex.is_a?(String) || message_or_regex.is_a?(Regexp)
+          raise ArgumentError, "Expected message_or_regex to be a String or a Regexp but was a #{message_or_regex.class.name}"
+        end
+        allowed_warnings << message_or_regex
+      end
+
+      def after_all_callback
+        @after_all_callback ||= -> { report_unexpected_warnings }
+      end
+
+      def report_unexpected_warnings
+        unexpected_warnings.empty? or warn <<~EOS.chomp
+          =====
+          Unexpected Deprecation Warnings Encountered
+            #{unexpected_warnings.to_a.join("\n  ")}
+          =====
+        EOS
+      end
+
+      def track_warning(warning)
+        if warning && !expected_warning?(warning)
+          unexpected_warnings << warning
+        end
+      end
+
+      protected
+
+      def unexpected_warnings
+        @unexpected_warnings ||= Set.new
+      end
+
+      def allowed_warnings
+        @allowed_warnings ||= Set.new
+      end
+
+      def expected_warning?(warning)
+        allowed_warnings.any? do |allowed_warning|
```

Comment:
Or maybe
```
allowed_warnings.any? { |allowed_warning| allowed_warning === warning }
```
?

Reply:
TIL about `===` doing exact string match and regex match all in one.  That seems very confusing because `===` is usually a more exact match than `==` rather than a fuzzier match isn't it?

Reply:
That sounds like the Javascript meaning of `===`. In Ruby, `===` means "run the `case when` matcher". Here's another example:
```
0...10 === 9
=> true
```
I like it, but I think Hound doesn't. So it's up to you.

Reply:
Hound seems to be fine with it and it reads easier, so it's all switched over.